src/mpl/configure.ac

```diff
@@ -684,7 +684,31 @@ AC_DEFINE_UNQUOTED([THREAD_PACKAGE_NAME],[$THREAD_PACKAGE_NAME],[set to the name
 
 # check for compiler-support for thread-local storage (MPL_TLS)
 if test "${with_thread_package}" != "argobots" ; then
-    AX_TLS([], AC_MSG_WARN([thread-local storage not supported by the compiler.]))
+    AC_MSG_CHECKING([for _Thread_local])
+    AC_LINK_IFELSE(
+        [AC_LANG_PROGRAM([extern _Thread_local int foo = 0;],[foo = 1])],
+	[pac_cv_have__thread_local=yes], [pac_cv_have__thread_local=no])
+    AC_MSG_RESULT([$pac_cv_have__thread_local])
+    AC_MSG_CHECKING([for __thread])
+    AC_LINK_IFELSE(
+        [AC_LANG_PROGRAM([__thread int foo = 0;],[foo = 1])],
+	[pac_cv_have___thread=yes],[pac_cv_have___thread=no])
+    AC_MSG_RESULT([$pac_cv_have___thread])
+    AC_MSG_CHECKING([for __declspec(thread)])
+    AC_LINK_IFELSE(
+        [AC_LANG_PROGRAM([__declspec(thread) int foo = 0;],[foo = 1])],
+	[pac_cv_have___declspec_thread]=yes,[pac_cv_have___declspec_thread=no])
```

Comment:
Misplaced closing `]`?

Reply:
Since we are only going to use one TLS definition, can we do `elif` or `if -z $pac_cv_tls ...` (assuming we'll set pac_cv_tls to the first keyword that's available)?

Reply:
I initially tried nesting the `AC_LINK_IFELSE` macros so they would only run if the previous one failed, but it was kind of hard to follow. I'm open to changing the logic here, but I'm not sure I'm getting the suggestion. Can you put it in pseudo-code? Or just add a commit to this branch?

Reply:
Hmm, maybe the nesting isn't too bad. New patch coming.